felix/dataplane/linux/endpoint_mgr.go

```diff
@@ -707,16 +708,32 @@ func (m *endpointManager) resolveWorkloadEndpoints() {
 
 	m.wlIfaceNamesToReconfigure.Iter(func(item interface{}) error {
 		ifaceName := item.(string)
-		err := m.configureInterface(ifaceName)
+
+		exists, err := m.interfaceExistsInProcSys(ifaceName)
 		if err != nil {
-			if exists, err := m.interfaceExistsInProcSys(ifaceName); err == nil && !exists {
-				// Suppress log spam if interface has been removed.
-				log.WithError(err).Debug("Failed to configure interface and it seems to be gone")
-			} else {
-				log.WithError(err).Warn("Failed to configure interface, will retry")
-			}
+			log.WithError(err).WithField("ifaceName", ifaceName).Warn("Failed to check if interface exists, will retry")
+			return nil
+		}
+		if !exists {
+			// Suppress log spam if interface has been removed.
+			log.WithField("ifaceName", ifaceName).Debug("Interface seems to be gone, will retry")
+			return nil
+		}
+
+		writable, err := m.interfaceWritableInProcSys(ifaceName)
+		if err != nil {
+			log.WithError(err).WithField("ifaceName", ifaceName).Warn("Failed to check if interface is writeable, will retry")
 			return nil
 		}
+		if writable {
+			if err := m.configureInterface(ifaceName); err != nil {
+				log.WithError(err).WithField("ifaceName", ifaceName).Warn("Failed to configure interface, will retry")
+				return nil
+			}
+		} else {
+			log.WithField("ifaceName", ifaceName).Info("Skipping configure for non-writeable interface.")
```

Comment:
If we're running in non-privileged mode, we'll see this message for every container interface which might balloon our logs a bit too much especially since the Felix interface configuration isn't expected to work in non-privileged mode. Maybe we should change the log level to `Debug` instead?

Reply:
Thanks for reviewing. You are probably right as i'm not familiar with the code. My assumption was that if the code doesn't exit early with "return nil" it would only run once for new devices. If that's not the case it would indeed end up ballooning the logs end setting it to debug would be better.

Reply:
I can't remember all of the cases that would call this off the top of my head, but I was thinking more about this from the side of all of the cali interfaces we create for the pods. If you have a ton of pods coming and going, then you'll see a lot of logs for those pods which might not be super useful since we know that in non-privileged mode, Felix doesn't have the permissions to write these values.

Reply:
Make sense. I'll update the PR!

Reply:
Done.